pkg/controller/provider/container/vsphere/collector.go

```diff
@@ -495,11 +498,11 @@ func (r *Collector) watch() (list []*libmodel.Watch) {
 
 //
 // Build the client.
-func (r *Collector) connect(ctx context.Context) error {
+func (r *Collector) connect(ctx context.Context) (status int, err error) {
 	r.close()
 	url, err := liburl.Parse(r.url)
 	if err != nil {
-		return liberr.Wrap(err)
+		return 0, liberr.Wrap(err)
```

Comment:
wouldn't it be better to set `err` and return? I mean, there's no http error code 0 but still look a bit awkward..